src/ProtoBuf.FSharp/CodeGen.fs

```diff
@@ -1,13 +1,44 @@
 module internal ProtoBuf.FSharp.CodeGen
 
 open System
-open FSharp.Reflection
 open System.Collections.Concurrent
+open System.Collections.Generic
+open System.Threading
 open System.Reflection
 open System.Reflection.Emit
+open FSharp.Reflection
 open MethodHelpers
 
-let inline emitZeroValueOntoEvaluationStack (gen: ILGenerator) (getterType: MethodType) =
+
+type private TypeBuilder with
+    member tb.DefineOpExplicit(src : Type, dst : Type) =
+        let attr = MethodAttributes.Public ||| MethodAttributes.HideBySig ||| MethodAttributes.SpecialName ||| MethodAttributes.Static
+        tb.DefineMethod("op_Explicit", attr, dst, [| src |])
+
+    member tb.SetProtoContractAttribute(skipConstructor : bool) =
+        let t = typeof<ProtoBuf.ProtoContractAttribute>
+        CustomAttributeBuilder(
+            t.GetConstructor [||], [||]
+            , [| t.GetProperty "ImplicitFields" ; t.GetProperty "SkipConstructor" |]
+            , [| box ProtoBuf.ImplicitFields.AllFields ; box skipConstructor |]
+        ) |> tb.SetCustomAttribute
+
+    member tb.DefineFieldForProtobuf(fi : PropertyInfo) =
+        tb.DefineField(fi.Name, fi.PropertyType, FieldAttributes.Public) // Do something with name and attributes?
+
+type private FieldBuilder with
+    member fb.SetProtoMemberAttribute(tag : int) =
+        let t = typeof<ProtoBuf.ProtoMemberAttribute>
+        CustomAttributeBuilder(t.GetConstructor [| typeof<int> |], [| box tag |]) |> fb.SetCustomAttribute
+
+
+let private emitDefaultValueViaCell (gen : ILGenerator) (tp : Type) =
+    let cell = gen.DeclareLocal(tp)
+    gen.Emit(OpCodes.Ldloca_S, cell)
+    gen.Emit(OpCodes.Initobj, tp)
+    gen.Emit(OpCodes.Ldloc, cell)
+
+let private emitZeroValueOntoEvaluationStack (gen: ILGenerator) (getterType: MethodType) =
```

Comment:
Description here.

Reply:
Well, I'd say that's the only one with reasonably good name :)